### PR TITLE
Optionally enforce required elements in XSDs

### LINKED ISSUE #162: Required fields lead to non required element in XSD
E.g. the required field `prefix` from https://github.com/apache/maven/blob/master/maven-repository-metadata/src/main/mdo/metadata.mdo#L354-L360 leads to this XSD:
```
<xs:element minOccurs="0" name="prefix" type="xs:string">
```

Obsviously the element with name `prefix` must leave out minOccurs="0" to be required in the XMLs.

### PR DESCRIPTION
**Pull request by kwin**
This closes #162

### CODE REVIEW

**Review by michael-o — APPROVED** [review]

File: modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java [hunk 1]
```diff
@@ -45,6 +45,15 @@
...
     private File outputDirectory;
 
+    /**
+     * Boolean flag relaxing XSD generation with regards to mandatory elements.
+     * If set to {@code true} will not require mandatory elements in the XML which can be useful if the XML is post processed (e.g. POM merging with parents).
+     * The default value is true for backwards compatibility reasons, but should be set to false for most cases.
+     * @since 2.1.0
+     */
+    @Parameter( defaultValue = "true")
+    private boolean areMandatoryElementsNotEnforced;
```
Comment (gnodet) [inline modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java:55]: This does not look very usual to have a boolean property starting with `is` or `are`.
Usually, the getters are written `isXxx` but the field is most often named without the `is`/`are` prefix.
I would propose to invert it and name it `enforceMandatoryElements` and default it to `false`.

**Review by gnodet — CHANGES_REQUESTED** [review]

File: modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java (hunk 1 above)
Comment (kwin) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java:55]: I can rename but the default should be to enforce mandatory elements. The POM case is really an unusual edge case IMHO

Comment (gnodet) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java:55]: I may have missed something, but the current default value is `true` for _not enforcing mandatory elements_.  I'm fine with changing the default value though.

Comment (gnodet) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java:55]: The javadoc should be changed accordingly if the default value is to be changed.

Comment (kwin) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java:55]: Sorry, my bad. I will think again about defaults.

Comment (kwin) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java]: Fixed as you proposed @gnodet in https://github.com/codehaus-plexus/modello/pull/263/commits/9ba65d849e1afaa7449980c2907900341dde816f.

Comment (kwin) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java]: Lets start with the default being no enforcement for backwards-compatibility.

**Pull request merged**